code/modules/mob/living/simple_animal/hostile/headslug.dm

```diff
@@ -43,32 +41,37 @@
 		I.forceMove(egg)
 	visible_message("<span class='warning'>[src] plants something in [victim]'s flesh!</span>", \
 					"<span class='danger'>We inject our egg into [victim]'s body!</span>")
-	egg_lain = 1
+	egg_layed = TRUE
 
-/mob/living/simple_animal/hostile/headslug/AttackingTarget()
-	. = ..()
-	if(. && !egg_lain && iscarbon(target) && !issmall(target))
-		// Changeling egg can survive in aliens!
-		var/mob/living/carbon/C = target
-		if(C.stat == DEAD)
-			if(HAS_TRAIT(C, TRAIT_XENO_HOST))
-				to_chat(src, "<span class='userdanger'>A foreign presence repels us from this body. Perhaps we should try to infest another?</span>")
-				return
-			Infect(target)
-			to_chat(src, "<span class='userdanger'>With our egg laid, our death approaches rapidly...</span>")
-			addtimer(CALLBACK(src, PROC_REF(death)), 10 SECONDS)
+/mob/living/simple_animal/hostile/headslug/AltClickOn(mob/living/carbon/carbon_target)
+	if(egg_layed || !istype(carbon_target) || !Adjacent(carbon_target))
+		return ..()
+	if(carbon_target.stat != DEAD && !do_mob(src, carbon_target, 5 SECONDS))
+		return
+	if(HAS_TRAIT(carbon_target, TRAIT_XENO_HOST))
+		to_chat(src, "<span class='userdanger'>A foreign presence repels us from this body. Perhaps we should try to infest another?</span>")
+		return
+	Infect(carbon_target)
+	to_chat(src, "<span class='userdanger'>With our egg laid, our death approaches rapidly...</span>")
+	addtimer(CALLBACK(src, PROC_REF(death)), 25 SECONDS)
 
 /obj/item/organ/internal/body_egg/changeling_egg
 	name = "changeling egg"
 	desc = "Twitching and disgusting."
 	origin_tech = "biotech=7" // You need to be really lucky to obtain it.
 	var/datum/mind/origin
-	var/time
+	var/time = 0
 
 /obj/item/organ/internal/body_egg/changeling_egg/egg_process()
-	// Changeling eggs grow in dead people
-	time++
```

Comment:
was time++ meant to be removed here? I dont see anywhere else it increases

Reply:
No that was just an accidental deletion at the end of me sorting things around